waku/v2/utils/limiter.go

```diff
@@ -0,0 +1,72 @@
+package utils
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/jellydator/ttlcache/v3"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"golang.org/x/time/rate"
+)
+
+type RateLimiter struct {
+	sync.Mutex
+	limiters *ttlcache.Cache[peer.ID, *rate.Limiter]
+	r        rate.Limit
+	b        int
+}
+
+func NewRateLimiter(r rate.Limit, b int) *RateLimiter {
+	return &RateLimiter{
+		r: r,
+		b: b,
+		limiters: ttlcache.New[peer.ID, *rate.Limiter](
+			ttlcache.WithTTL[peer.ID, *rate.Limiter](30 * time.Minute),
```

Comment:
It shouldnt be a problem I think? because if for some reason you are connected to that peer, exceed the 30m and do a new request, we'd create a rate limiter from scratch, altho I imagine that it would be rare for a filter peer to remain connected for so long without doing any request? At least ping requests should happen, and ttlcache does extend the expiration date when you call a `Get`.